src/services/completions.ts

```diff
@@ -1028,6 +1054,24 @@ namespace ts.Completions {
         return createCompletionDetails(name, ScriptElementKindModifier.none, kind, [displayPart(name, kind2)]);
     }
 
+    function createImportMetaDetails(node: MetaProperty): CompletionEntryDetails {
+        return {
+            name: getMetaPropertyName(node),
+            kind: ScriptElementKind.memberVariableElement,
+            kindModifiers: ScriptElementKindModifier.none,
+            displayParts: [
+                displayPart("(", SymbolDisplayPartKind.punctuation),
+                displayPart("property", SymbolDisplayPartKind.text),
+                displayPart(")", SymbolDisplayPartKind.punctuation),
+                displayPart(" ", SymbolDisplayPartKind.space),
+                displayPart(getMetaPropertyName(node), SymbolDisplayPartKind.propertyName),
+                displayPart(":", SymbolDisplayPartKind.punctuation),
+                displayPart(" ", SymbolDisplayPartKind.space),
+                displayPart("ImportMeta", SymbolDisplayPartKind.interfaceName),

```

Comment:
I copied this from the current output... what would be better?

Reply:
You should be able to get a reference to the `ImportMeta` global interface symbol and use the standard symbol display stuff. Probably the easiest way with the current (internal) API is `checker.resolveName` with an undefined location, though it might be reasonable to expose `getGlobalImportMetaType` internally, and I assume the `ImportMeta` interface symbol is exposed on that type.

Reply:
@andrewbranch sorry, I was just away camping for the last 10 days, but I'm back now.

Opps... I messed up this code initially as this should be showing, I believe, `NewableFunction` for `event.target`. That should be fixed now.

I'm not sure it helps to call `resolveName` providing `ImportMeta` because that only gets the symbol of `ImportMeta`, but that name could just be displayed here.

Reply:
I guess I don’t understand how it ever worked before:

> I copied this from the current output...

This got produced from the dummy symbol?

Reply:
I copied it from the output of the original commit change that was wrong way of doing things. That would return the `ImportMeta` type from the dummy symbol: https://github.com/microsoft/TypeScript/pull/45031/commits/f9ded349d7602ff37ded1e38380fe419d60c60f2

Reply:
I see. I guess it would be nice if `symbolDisplay.ts` had a function that could take a name and a type and produce this output.

Reply:
Sounds good! I'll do that tomorrow and thanks for the help.

By the way, isn't there a way we could just make this work by having some `import` and `new` symbol similar to the `arguments` symbol then add `meta` and `target` properties respectively? I don't understand symbols or the compiler internals well enough to know how this all works or if that would cause problems.

Reply:
That seems totally plausible to me; not sure if there’s some reason why it’s not implemented like that. I’d be open to that, but would want to bring in a couple other reviewers to help think about edge case implications of it.